velox/exec/HashProbe.cpp

```diff
@@ -167,12 +167,15 @@ void HashProbe::initialize() {
 
   size_t numIdentityProjections = 0;
   for (auto i = 0; i < probeType_->size(); ++i) {
-    auto name = probeType_->nameOf(i);
+    auto& name = probeType_->nameOf(i);
     auto outIndex = outputType_->getChildIdxIfExists(name);
-    if (outIndex.has_value()) {
-      projectedInputColumns_.insert(i);
-      identityProjections_.emplace_back(i, outIndex.value());
-      if (outIndex.value() == i) {
+    if (!outIndex.has_value()) {
+      continue;
+    }
+    projectedInputColumns_[i] = *outIndex;
+    if (!isRightJoin(joinType_) && !isFullJoin(joinType_)) {
+      identityProjections_.emplace_back(i, *outIndex);
```

Comment:
Is this right to add this entry unconditionally?
Below we `++numIdentityProjections` only if indices are the same.
Looks like `identityProjections_` stores pairs regardless of the condition, so is it named right?

Reply:
We do have check `!isRightJoin(joinType_) && !isFullJoin(joinType_)`.  If the join is not RIGHT or FULL, the probe side columns are indeed identity projection so this is correct.

Reply:
Oh I see what you mean, the usage of `numIdentityProjections` is a little confusing, it is actually used to track if the columns are identity project and if the column order in the output type is also the exactly same as probe input.